src/historywork/WriteVerifiedCheckpointHashesWork.cpp

```diff
@@ -182,13 +262,51 @@ WriteVerifiedCheckpointHashesWork::endOutputFile()
 {
     if (mOutputFile && mOutputFile->is_open())
```

Comment:
We should probably throw if `mOutputFile && mOutputFile->is_open()` is false, instead of silently suceeding without writing an output file.

Reply:
`endOutputFile` is called by `resetIter` (which then calls start output file, so no problem there), `~ WriteVerifiedCheckpointHashesWork`, and `onSuccess`. When the work finishes, `onSuccess` executes and the file is written to and closed. Then later when the shared ptr containing the work is reset `~ WriteVerifiedCheckpointHashesWork` is called and  `mOutputFile` is not open.  (in this diff, an spurious warning will be emitted). I think instead we should not be emitting a warning or raising an error in that case. I agree that we shouldn't silently succeed without writing an output file, but Considering there is logic to report errors opening the file in `startOutputFile`, I don't think we need to warn or throw here.